source/javascript/src/components/CollectionItem.js

```diff
@@ -174,6 +247,45 @@ class CollectionItem extends HTMLElement {
   }
 }
 
+/**
+ * Update given collection to have given name.
+ * @author William Chung <wchung@ucsd.edu>
+ * @param prevName Name of the collection to be udpated.
+ * @param newName New name of the collection .
+ */
+function updateCollectionName (prevName, newName) {
+  wrapper.transaction((event) => {
+    const db = event.target.result
+
+    const transaction = db.transaction(['currentLogStore'], 'readwrite')
+    const objectStore = transaction.objectStore('currentLogStore')
+    objectStore.openCursor().onsuccess = function (event) {
+      const cursor = event.target.result
+      if (cursor) {
+        const json = cursor.value
+
+        const collections = json.properties.collections
+
+        collections.forEach(collection => {
+          if (collection.name === prevName) {
+            collection.name = newName
+          }
+        })
+
+        // Save changes
+        const requestUpdate = cursor.update(json)
+        requestUpdate.onerror = function (event) {
+          // Do something with the error
+        }
+        requestUpdate.onsuccess = function (event) {
+          // Success - the data is updated!
+          console.log('successfully updated "' + prevName + '" to "' + newName + '"')
+        }
+      }
+    }
+  })
+}
+
 customElements.define('collection-item', CollectionItem)
 
-export { CollectionItem }
+export { CollectionItem, wrapper }
```

Comment:
did wrapper not work unless it was included in the exports block?

Reply:
I noticed we were creating a new wrapper object each time we needed to a make transaction so I decided to make one to share between the files. I don't know if this is best practice though.

Reply:
Sounds similar to a singleton design pattern https://www.sitepoint.com/javascript-design-patterns-singleton/; I'd say this is a good idea if we have enough time at the end of the sprint